modules/github_integration/lib/api/v3/github_integration/github_pull_request_representer.rb

```diff
@@ -0,0 +1,166 @@
+
+        property :repository,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        date_time_property :github_updated_at,
+                           render_nil: true,
+                           setter: ->(*) { nil }
+
+        property :title,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :body,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :draft,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :merged,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :merged_at,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :comments_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :review_comments_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :additions_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :deletions_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :changed_files_count,
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :labels,
+                 setter: ->(*) { nil }
+
+        property :github_user,
+                 getter: ->(*) {
+                   {
+                     login: github_user.github_login,
+                     htmlUrl: github_user.github_html_url,
+                     avatarUrl: github_user.github_avatar_url
+                   }
+                 },
+                 setter: ->(*) { nil }
+
+        property :merged_by,
+                 getter: ->(*) {
+                   next unless merged_by
+
+                   {
+                     login: merged_by.github_login,
+                     htmlUrl: merged_by.github_html_url,
+                     avatarUrl: merged_by.github_avatar_url
+                   }
+                 },
+                 render_nil: true,
+                 setter: ->(*) { nil }
+
+        property :github_check_runs,
+                 getter: ->(*) {
+                   latest_check_runs.map do |check_run|
+                     {
+                       htmlUrl: check_run.github_html_url,
+                       appOwnerAvatarUrl: check_run.github_app_owner_avatar_url,
+                       name: check_run.name,
+                       status: check_run.status,
+                       conclusion: check_run.conclusion,
+                       outputTitle: check_run.output_title,
+                       outputSummary: check_run.output_summary,
+                       detailsUrl: check_run.details_url,
+                       startedAt: check_run.started_at&.iso8601,
+                       completedAt: check_run.completed_at&.iso8601
+                     }
```

Comment:
Also feels like a resource of its own.

Reply:
yeeesss, and i somehow always land in the same problem - all things need to be scope under a work package (to get permissions right - I need to have a `project` to see whether the `current_user` can `:show_github_content`. But when I have an endpoint like `/api/v3/github_users/<id>` this is not enforceable (even worse for `check_runs` or `pull_requests` as these have actual private repo data in them).